rust-keybroker/keybroker-server/src/challenge.rs

```diff
@@ -70,27 +75,43 @@ impl Challenger {
         &mut self,
         key_id: &str,
         wrapping_key: &PublicWrappingKey,
+        mock_challenge: bool,
     ) -> Challenge {
         // All challenges are given random u32 identities
-        let mut challenge_id: u32 = rand::random();
+        let mut challenge_id: u32 = self.rng.gen();
 
         // Simple lightweight collision avoidance - probably not needed given u32 distribution space,
         // but it's easy to do. Also allows us to throw out zero if we get it.
         while challenge_id == 0 || self.challenge_table.contains_key(&challenge_id) {
-            challenge_id = rand::random();
+            challenge_id = self.rng.gen();
         }
 
         let challenge = Challenge {
             challenge_id,
             key_id: key_id.to_owned(),
             wrapping_key: wrapping_key.clone(),
-            // TODO: We should create a random nonce here. The use of this mock value allows the
-            // server to be tested with a hard-coded example attestation token during development.
-            challenge_value: CCA_EXAMPLE_TOKEN_NONCE.to_vec(),
+            challenge_value: if mock_challenge {
+                CCA_EXAMPLE_TOKEN_NONCE.to_vec()
+            } else {
+                let mut v: Vec<u8> = vec![0; 64];
+                self.rng.fill(&mut v[..]);
+                v
+            },
         };
 
         self.challenge_table.insert(challenge_id, challenge.clone());
 
+        if self.verbose {
+            println!("Created challenge:");
```

Comment:
If we are logging more events in the server, we should probably move away from doing it with `println!` and use a more configurable logging mechanism instead. But again, no need to address that within this PR. Just a thought for the future.

Reply:
Aggreed. We can improve logging later on --- and make it consistent with the other parts of the whole project as I'm sure logging will be required elsewhere.